datastore/additional_data/models.py

```diff
@@ -81,3 +81,14 @@ def __str__(self):
             return "%s - %s" % (self.data["name"], self.org_id)
         else:
             return "%s" % self.org_id
+
+
+class NSPL(models.Model):
+    # Postcode without whitespaces and uppercase.
+    postcode = models.CharField(max_length=7)
```

Comment:
Not sure how you do it in django, but this postcode field definitely needs an index - the lookups are very slow without it. Potentially a unique index as each postcode should be different.

Reply:
In Django an `id` field is added automatically.

Reply:
But because the `postcode` field is used for the lookup, which is unindexed, it's very slow. It looks like adding `db_index=True` to the field definition will add a postgres index on the `postcode` field (while `id` will stay the primary key) https://docs.djangoproject.com/en/3.0/ref/models/fields/#db-index

Reply:
I have added `db_index=True` in `postcode`.